src/Data/Models/FMUTXO.cs

```diff
@@ -55,6 +77,12 @@ public bool Equals(FMUTXO? other)
             return TxId == other.TxId && OutputIndex == other.OutputIndex && SatsAmount == other.SatsAmount;
         }
 
+        public bool Equals(UTXO? other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            return TxId == other.Outpoint.Hash.ToString() && OutputIndex == other.Outpoint.N && SatsAmount == ((Money)other.Value).Satoshi;
```

Comment:
is `SatsAmount` needed here? I think if the hash and index are the same, then we're talking about the same utxo

Reply:
You are right, It's probably not needed, but it's a good check anyway. Also, it is checked in the `public bool Equals(FMUTXO? other)` method.